app/lib/dartdoc/models.dart

```diff
@@ -3,16 +3,112 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart:async';
+import 'dart:convert';
 
 import 'package:json_annotation/json_annotation.dart';
 import 'package:meta/meta.dart';
 
+import '../shared/datastore.dart' as db;
 import '../shared/utils.dart' show jsonUtf8Encoder, utf8JsonDecoder;
 import '../shared/versions.dart' as versions;
 import 'storage_path.dart' as storage_path;
 
 part 'models.g.dart';
 
+/// Status values for [DartdocRecord].
+abstract class DartdocRecordStatus {
+  static const uploading = 'uploading';
+  static const ready = 'ready';
+  static const deleting = 'deleting';
+}
```

Comment:
Not sure, why we need `deleting`, maybe we just need a boolean `ready` (that also makes our model simpler)

Reply:
It may be needed (not sure about it yet) by the serving part: if an entry becomes deleting, we shouldn't serve from that content anymore.